pkg/kubectl/cmd/cmd.go

```diff
@@ -303,6 +304,49 @@ func NewDefaultKubectlCommand() *cobra.Command {
 	return NewDefaultKubectlCommandWithArgs(NewDefaultPluginHandler(plugin.ValidPluginFilenamePrefixes), os.Args, os.Stdin, os.Stdout, os.Stderr)
 }
 
+func hasNoOptDefVal(name string, fs *pflag.FlagSet) bool {
+	flag := fs.Lookup(name)
+	if flag == nil {
+		return false
+	}
+	return flag.NoOptDefVal != ""
+}
+
+func shortHasNoOptDefVal(name string, fs *pflag.FlagSet) bool {
+	if len(name) == 0 {
+		return false
+	}
+
+	flag := fs.ShorthandLookup(name[:1])
+	if flag == nil {
+		return false
+	}
+	return flag.NoOptDefVal != ""
+}
+
+func appendPersistentFlagsAfterCommand(args []string, c *cobra.Command) []string {
+	flags := c.PersistentFlags()
+	for i := 0; i < len(args); i++ {
+		s := args[i]
+		switch {
+		case strings.HasPrefix(s, "--") && !strings.Contains(s, "=") && !hasNoOptDefVal(s[2:], flags):
+			// If '--flag arg' then jump past flag and arg
```

Comment:
arg is optional for boolean flag, so I don't think you can make this assumption.  For example `--all-namespaces`